src/Client/wwwroot/index.html

```diff
@@ -86,6 +86,24 @@
         for (var i = 0; i < elements.length; i++) {
             elements[i].style.backgroundImage = 'linear-gradient(-120deg,' + primaryColor + ' 50%,' + baseColor + ' 50%)';
         }
+
+        async function downloadFileFromStream(fileName, contentStreamReference) {
```

Comment:
Can JS be avoided?

Reply:
This is actually the recommended way by microsoft: https://docs.microsoft.com/en-us/aspnet/core/blazor/file-downloads?view=aspnetcore-6.0 (and as far as I know the only way to do it in blazor)